lib/jekyll/static_file.rb

```diff
@@ -97,6 +97,7 @@ def write(dest)
 
     def to_liquid
       {
+        "name"          => File.basename(name, extname),
```

Comment:
I'm fine with adding this to Liquid. When you do `static_file.name`, it expects something else, though?